python/paddle/nn/layer/loss.py

```diff
@@ -13,9 +13,93 @@
 # limitations under the License.
 
 # TODO: define loss functions of neural network  
-# __all__ = ['NCELoss',
-#            'CrossEntropyLoss',
-#            'MSELoss',
-#            'L1Loss',
-#            'NLLLoss',
-#            'BCELoss']
+import paddle.fluid as fluid
+__all__ = [
+    #'NCELoss',
+    #    'CrossEntropyLoss',
+    #    'MSELoss',
+    'L1Loss',
+    #    'NLLLoss',
+    #   'BCELoss'
+]
+
+
+class L1Loss(fluid.dygraph.Layer):
+    """
+    This interface is used to construct a callable object of the ``L1Loss`` class.
+    The L1Loss layer calculates the mean absolute error of input predictions and target label.
+    For input predictions label, and target label, the loss is calculated as follows.
+    If :attr:`reduction` set to ``'none'``, the unreduced loss is:
+    .. math::
+        Out = |input - label|
+    If :attr:`reduction` set to ``'mean'``, the reduced mean loss is:
+    .. math::
+        Out = MEAN(|input - label|)
+    If :attr:`reduction` set to ``'sum'``, the reduced sum loss is:
+    .. math::
+        Out = SUM(|input - label|)
+    Parameters:
+        reduction (str, optional): Indicate how to average the loss by batch_size, 
```

Comment:
By `batch_size` ? Please make sure that whether this API returns tensor with shape [1] or shape [1, other_dim].

Reply:
Thanks, it should return loss with shape [1]  when reduction is 'sum' or 'mean', and return loss with the same shape of input when reduction is 'none'.